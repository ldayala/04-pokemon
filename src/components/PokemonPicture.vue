<template>
  <div class="pokemon-container">
    <img :src="imgSrc" alt="pokemon" class="hidden-pokemon" />
    <img
      :src="imgSrc"
      alt="pokemon"
      class="fade-in"
      v-if="showPokemon"
    />
  </div>
</template>

<script>
export default {
  //number , required
  props:{
    pokemonId:{
      type:Number,
      required:true
    },
    showPokemon:false
  },
  computed: {
    imgSrc() {
      return `https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/${this.pokemonId}.svg`;
    },
  },
};
</script>

<style scoped>
/* Pokemon Picture */
.pokemon-container {
  height: 200px;
  display: flex;
  justify-content: center;
}
img {
  height: 200px;
  position: absolute;

  user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  -webkit-user-drag: none;
  -webkit-user-select: none;
}
.hidden-pokemon {
  filter: brightness(0);
}
</style>