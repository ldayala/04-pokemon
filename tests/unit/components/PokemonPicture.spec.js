import { shallowMount } from "@vue/test-utils";
import PokemonPicture from "@/components/PokemonPicture";
describe('PokemonPicture componet', () => {
    
    test('Debe hacer match con el snapshot', () => {
        const wrapper= shallowMount(PokemonPicture,{
            props:{
                pokemonId:1,
                showPokemon:false
            }
        })
        expect(wrapper.html).toMatchSnapshot()
    });


    test('Debe de mostrar la imagen oculta y el pokemon 100', () => {
        const wrapper= shallowMount(PokemonPicture,{
            props:{
                pokemonId:100,
                showPokemon:false
            }
        })
        const[img1,img2]= wrapper.findAll('img')
        expect(img1.exists()).toBeTruthy()

        expect(img2).toBe(undefined)

        //verificamos que la imagen una tenga la clase hidden-pokemon
        expect(img1.classes('hidden-pokemon')).toBe(true)
        expect(img1.attributes().src).toBe('https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/100.svg');
        
    });

    test('Debe mostrar la imagen del pokemon si showPokemon:true', () => {
        const wrapper= shallowMount(PokemonPicture,{
            props:{
                pokemonId:100,
                showPokemon:true
            }
        })
        const[img1,img2]= wrapper.findAll('img')
        
        expect(img1.exists()).toBeTruthy()

        expect(img2.exists()).toBeTruthy()

        //verificamos que la imagen una tenga la clase hidden-pokemon
        expect(img2.classes('fade-in')).toBe(true)
    });
});