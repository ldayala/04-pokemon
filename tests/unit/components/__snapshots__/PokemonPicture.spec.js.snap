// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonPicture componet Debe hacer match con el snapshot 1`] = `[Function]`;
